platforms/documentation/docs/src/docs/userguide/dep-man/declaring-repositories/supported_repository_protocols.adoc

```diff
@@ -0,0 +1,332 @@
+Digest access authentication over HTTP.
+
+link:{javadocPath}/org/gradle/authentication/http/HttpHeaderAuthentication.html[HttpHeaderAuthentication]::
+Authentication based on a custom HTTP header, such as private tokens or OAuth tokens.
+
+[[sub:preemptive_authentication]]
+=== Using preemptive authentication
+
+By default, Gradle submits credentials only when a server responds with an authentication challenge (HTTP 401).
+However, some servers might respond with a different code (e.g., GitHub returns a 404) that could cause dependency resolution to fail.
+In such cases, you can configure Gradle to send credentials preemptively by explicitly using the link:{javadocPath}/org/gradle/authentication/http/BasicAuthentication.html[BasicAuthentication] scheme:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=preemptive-authentication]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=preemptive-authentication]"]
+====
+
+[[sub:http-header-auth]]
+=== Using HTTP header authentication
+
+For Maven repositories that require token-based, OAuth2, or other HTTP header-based authentication, you can use link:{javadocPath}/org/gradle/api/credentials/HttpHeaderCredentials.html[HttpHeaderCredentials] and link:{javadocPath}/org/gradle/authentication/http/HttpHeaderAuthentication.html[HttpHeaderAuthentication]:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=header-authenticated-maven-repo]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=header-authenticated-maven-repo]"]
+====
+
+[[sec:s3-repositories]]
+=== AWS S3 repositories configuration
+
+When configuring a repository that uses AWS S3, several options and settings are available.
+
+[[sub:s3_configuration_properties]]
+==== S3 configuration properties
+
+The following system properties can be used to configure interactions with S3 repositories:
+
+`org.gradle.s3.endpoint`::
+Overrides the AWS S3 endpoint when using a non-AWS, S3 API-compatible storage service.
+
+`org.gradle.s3.maxErrorRetry`::
+Specifies the maximum number of retry attempts when the S3 server responds with an HTTP 5xx status code. The default value is 3 if not specified.
+
+[[sub:s3_url_formats]]
+==== S3 URL formats
+
+S3 URLs must use the 'virtual-hosted-style' format:
+
+[source,text]
+----
+s3://<bucketName>[.<regionSpecificEndpoint>]/<s3Key>
+----
+
+*Example*: `s3://myBucket.s3.eu-central-1.amazonaws.com/maven/release`
+
+- `myBucket`: The AWS S3 bucket name.
+- `s3.eu-central-1.amazonaws.com`: The optional region-specific endpoint.
+- `/maven/release`: The AWS S3 key (a unique identifier for an object within a bucket).
+
+[[sub:s3_proxy_settings]]
+==== S3 proxy settings
+
+A proxy for S3 can be configured using the following system properties:
```

Comment:
It looks like a lot of this information is coming from `declaring_repositories_adv`. 

Should we be deleting the original content in this PR now to avoid duplication?

Reply:
I will have a separate PR for this, as some content has been moved to other PRs (not just this one). This ensures that content is not lost between releases if the other PRs aren't merged.